src/toxcore/dht/server.rs

```diff
@@ -758,38 +787,39 @@ mod tests {
         ], id: 38 };
 
         let nodes_resp = NodesResponse::new(&precomp, &bob_pk, nrs.clone());
-        let mut client = alice.create_client(&addr, bob_pk);
+        let mut state = alice.state.write();
+        let mut client = alice.create_client(&addr, bob_pk, &state.peers_cache);
         client.ping_id = 38;
-        alice.handle_nodes_resp(client, nodes_resp).wait().unwrap();
+        alice.handle_nodes_resp(client, nodes_resp, &mut state.kbucket).wait().unwrap();
         let mut kbuc = Kbucket::new(KBUCKET_BUCKETS, &alice.pk);
         for pn in &nrs.nodes {
             kbuc.try_add(pn);
         }
 
-        is_kbucket_eq(&alice, kbuc);
+        assert_eq!(state.kbucket, kbuc);
 
         // error case, can't decrypt
         let nodes_resp = NodesResponse::new(&precomp, &alice.pk, nrs.clone());
         let pk = alice.pk;
-        let mut client = alice.create_client(&addr, pk);
+        let mut client = alice.create_client(&addr, pk, &state.peers_cache);
         client.ping_id = 38;
-        assert!(alice.handle_nodes_resp(client, nodes_resp).wait().is_err());
+        assert!(alice.handle_nodes_resp(client, nodes_resp, &mut state.kbucket).wait().is_err());
         // ping_id = 0
         let nrs = NodesResponsePayload { nodes: vec![
             PackedNode::new(false, SocketAddr::V4("127.0.0.1:12345".parse().unwrap()), &gen_keypair().0)
         ], id: 0 };
         let nodes_resp = NodesResponse::new(&precomp, &bob_pk, nrs.clone());
-        let mut client = alice.create_client(&addr, bob_pk);
+        let mut client = alice.create_client(&addr, bob_pk, &state.peers_cache);
         client.ping_id = 0;
-        assert!(alice.handle_nodes_resp(client.clone(), nodes_resp.clone()).wait().is_err());
+        assert!(alice.handle_nodes_resp(client.clone(), nodes_resp.clone(), &mut state.kbucket).wait().is_err());
```

Comment:
You should test it by passing special packet and calling `handle_packet`, not by calling handle_YOUR_PACKET directly. See https://github.com/tox-rs/tox/blob/master/src/toxcore/tcp/server/server.rs#L430